### PR TITLE
add compile fix

### PR DESCRIPTION
**Pull request by joequant**
fixes these errors

/tmp/user/spack-stage/spack-stage-lccontent-3.1.5-s3ww3pfjecbqehfgqgxvydwcvqk466dp/spack-src/include/LCUtility/KDTreeLinkerAlgoT.h:275:22: error: '*' in boolean context, suggest '&&' instead [-Werror=int-in-bool-context]
  275 |             isInside *= thedim >= trackBox.dimmin[i] && thedim <= trackBox.dimmax[i];
      |             ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/user/spack-stage/spack-stage-lccontent-3.1.5-s3ww3pfjecbqehfgqgxvydwcvqk466dp/spack-src/include/LCUtility/KDTreeLinkerAlgoT.h:292:30: error: '*' in boolean context, suggest '&&' instead [-Werror=int-in-bool-context]
  292 |             isFullyContained *= (regionmin >= trackBox.dimmin[i] && regionmax <= trackBox.dimmax[i]);
      |             ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/user/spack-stage/spack-stage-lccontent-3.1.5-s3ww3pfjecbqehfgqgxvydwcvqk466dp/spack-src/include/LCUtility/KDTreeLinkerAlgoT.h:293:29: error: '*' in boolean context, suggest '&&' instead [-Werror=int-in-bool-context]
  293 |             hasIntersection *= (regionmin < trackBox.dimmax[i] && regionmax > trackBox.dimmin[i]);
      |             ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/user/spack-stage/spack-stage-lccontent-3.1.5-s3ww3pfjecbqehfgqgxvydwcvqk466dp/spack-src/include/LCUtility/KDTreeLinkerAlgoT.h:313:30: error: '*' in boolean context, suggest '&&' instead [-Werror=int-in-bool-context]
  313 |             isFullyContained *= (regionmin >= trackBox.dimmin[i] && regionmax <= trackBox.dimmax[i]);
      |             ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/user/spack-stage/spack-stage-lccontent-3.1.5-s3ww3pfjecbqehfgqgxvydwcvqk466dp/spack-src/include/LCUtility/KDTreeLinkerAlgoT.h:314:29: error: '*' in boolean context, suggest '&&' instead [-Werror=int-in-bool-context]
  314 |             hasIntersection *= (regionmin < trackBox.dimmax[i] && regionmax > trackBox.dimmin[i]);
      |             ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

### CODE REVIEW

**Comment by vvolkl** [comment]
It seems like a sensible warning, so I wouldn't like to disable it. Why not do as the compiler suggest and add a patch to change ` var *= x`  to `var = var && x`?

**Comment by joequant** [comment]
Will try to see if that works, and if it does, will submit patch upstream.

**Comment by joequant** [comment]
Changed patch to file compile warning

File: packages/lccontent/package.py [hunk 1]
```diff
@@ -23,14 +23,18 @@ class Lccontent(CMakePackage):
...
 
     def cmake_args(self):
         args = [
-                '-DCMAKE_MODULE_PATH=%s' % self.spec["pandorapfa"].prefix.cmakemodules,
-                "-DCMAKE_CXX_FLAGS=-std=c++17"]
+                '-DCMAKE_CXX_STANDARD=17',
+                '-DCMAKE_MODULE_PATH=%s' % self.spec["pandorapfa"].prefix.cmakemodules
+        ]
+        if self.spec.satisfies('%gcc@10:'):
+            args.append('-DCMAKE_CXX_FLAGS=-Wno-int-in-bool-context')
```
Comment (vvolkl) [inline packages/lccontent/package.py:37]: With the patch, this flag is no longer needed.

**Review by vvolkl — CHANGES_REQUESTED** [review]
Once the compiler flag is removed, this looks good to be merged.

File: packages/lccontent/package.py (hunk 1 above)
Comment (vvolkl) [inline packages/lccontent/package.py:37]:
```suggestion

```

File: packages/lccontent/package.py [hunk 2]
```diff
@@ -23,14 +23,16 @@ class Lccontent(CMakePackage):
     patch("path2.patch")
     patch("path3.patch")
     patch("path4.patch")
+    patch("bool-int.patch")
```
Comment (vvolkl) [inline packages/lccontent/package.py:26]:
```suggestion
    patch("bool-int.patch", when="@:3.1.5")
```

**Pull request merged**